varya/classes/class-varya-custom-colors.php

```diff
@@ -0,0 +1,310 @@
+<?php
+/**
+ * Varya Theme: Custom Colors Class
+ *
+ * This class is in charge of color customization via the Customizer.
+ *
+ * Each variable that needs to be updated is defined in the $varya_custom_color_variables array below.
+ * A customizer setting is created for each color, and custom CSS-variables are enqueued in the front and back end.
+ *
+ * @package WordPress
+ * @subpackage Varya
+ * @since 1.0.0
+ */
+
+class Varya_Custom_Colors {
+
+private $varya_custom_color_variables = array();
+
+	function __construct() {
+
+		/**
+		 * Define color variables
+		 */
+		$this->$varya_custom_color_variables = array(
+			array( '--global--color-primary', '#000000', 'Primary Color' ),
+			array( '--global--color-secondary', '#A36265', 'Secondary Color' ),
+			array( '--global--color-foreground', '#333333', 'Foreground Color' ),
+			array( '--global--color-background', '#FFFFFF', 'Background Color' ),
+			array( '--global--color-border', '#EFEFEF', 'Borders Color' )
+		);
+
+		/**
+		 * Register Customizer actions
+		 */
+		add_action( 'customize_register', array( $this, 'varya_customize_custom_colors_register' ) );
+
+		/**
+		 * Enqueue color variables for customizer & frontend.
+		 */
+		add_action( 'wp_enqueue_scripts', array( $this, 'varya_custom_color_variables' ) );
+
+		/**
+		 * Enqueue color variables for editor.
+		 */
+		add_action( 'enqueue_block_editor_assets', array( $this, 'varya_editor_custom_color_variables' ) );
+	}
+
+	/**
+	 * Auto-set color luminance for hover and selection colors
+	 *
+	 * Lightens/darkens a given colour (hex format), returning the altered colour in hex format.7
+	 *
+	 * @param str $fg_hex Colour as hexadecimal (with or without hash);
+	 * @percent float $percent Decimal ( 0.2 = lighten by 20%(), -0.4 = darken by 40%() )
+	 * @return str Lightened/Darkend colour as hexadecimal (with hash);
+	 */
+	function varya_color_midpoint( $fg_hex, $bg_hex ) {
```

Comment:
I _think_ this function can be removed since we now use `varya_color_blend_by_opacity()`, right?